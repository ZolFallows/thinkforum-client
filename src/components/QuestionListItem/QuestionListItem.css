
.Question_group {
    border-top: 1px solid #ececec;
    padding: 10px 10px 20px 10px;
}

.delete_btn {
    font-weight: 600;
    font-size: 0.8rem;
    border: 1px solid #ececec;
    margin: 10px 0;
    padding: 5px 10px;
    background: white;
    cursor: pointer;
    outline: none;
}

.delete_btn:hover, .delete_btn:focus{
    background: #f8f8f8;
    transition: ease-out 0.5s;
}

.Question_group h3,
.Question_group h3 a  {
    margin: 10px 0;
    font-size: 1rem;
    font-weight: 500;
    text-decoration: none;
}

.discussion_info {
    padding-top: 4px;
    font-size: 0.7rem;
    color: grey;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.tag {
    background: #ececec;
    padding: 1.5px 3px 4px 5px;
    margin: 2px;
    border-radius: 2px;
}
.user_name {
    color: gray;
}

.user_name a {
    text-decoration: none;
    color: black;
    font-weight: 700;
}

@media (min-width: 320px) and (max-width: 480px) {
    .discussion_info {
        font-size: 0.7rem;
        display: flex;
        flex-direction: column;
    }
}