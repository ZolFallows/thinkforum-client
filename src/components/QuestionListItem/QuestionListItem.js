import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import Moment from 'react-moment';
import './QuestionListItem.css'

export default class QuestionListItem extends Component {

    // renderToggleDelete(question) {
    //     const payload = TokenService.readJwtToken()
    //     return TokenService.hasAuthToken() && Number(payload.user_id) === Number(question.user.id) 
    //                 ? <div className="delete" onClick={this.handleDeleteQuestion}>Delete</div>
    //                 : <p>
    //                     <span><Link to={`/user/${question.user.id}`}>{question.user.full_name}</Link></span> 
    //                     {' - '} 
    //                     <span>{question.user.user_name}</span>
    //                   </p>
    // }

    render(){
        const { question, isUser } = this.props
        return (
            <div className="Question_group">
                <h3>
                    <Link to={`/questions/${question.id}`}>{question.title}</Link>
                </h3>
                {TokenService.hasAuthToken() && isUser
                    && TokenService.readJwtToken().user_id === question.user.id
                    ? <button className="delete" onClick={() => this.props.handleDelete(question.id)}>Delete</button>
                    : <p>
                        <span><Link to={`/user/${question.user.id}`}>{question.user.full_name}</Link></span> 
                        {' - '} 
                        <span>{question.user.user_name}</span>
                      </p>
                }
                <div>
                    <div className="Tags">
                            {question.tags.split(',').map(tag => <span key={tag} className="tag">{tag} </span>)}
                    </div>
                    <div>
                        <span><Moment fromNow>{question.date_created}</Moment></span>
                        {' - '}
                        <span>{question.number_of_answers}respond(s)</span>
                    </div>
                </div>
            </div>
        )
    }
}