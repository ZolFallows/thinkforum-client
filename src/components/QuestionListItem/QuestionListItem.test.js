import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import QuestionListItem from './QuestionListItem'

const sectionProp = {
                        "id": 2,
                        "title": "Another discussion",
                        "content": "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias.",
                        "tags": "javascript,react",
                        "date_created": "2019-03-27T20:27:19.481Z",
                        "user": {
                            "id": 2,
                            "user_name": "b.deboop",
                            "full_name": "Bodeep Deboop",
                            "date_created": "2019-03-27T20:27:19.481Z"
                        },
                        "number_of_answers": 1
                    }


describe(`QuestionListItem Component`, () => {
    it('Renders without crashing', () => {
        shallow(<QuestionListItem />)
      });

    it('renders empty given no QuestionListItem without errors', () => {
        const wrapper = shallow(<QuestionListItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders given QuestionListItem without errors', () => {
        const wrapper = shallow(<QuestionListItem question={sectionProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})