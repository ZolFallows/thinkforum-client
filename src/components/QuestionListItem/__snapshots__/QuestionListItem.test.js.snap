// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionListItem Component renders empty given no QuestionListItem without errors 1`] = `<Fragment />`;

exports[`QuestionListItem Component renders given QuestionListItem without errors 1`] = `
<Fragment>
  <div
    className="Question_group"
  >
    <h3>
      <Link
        replace={false}
        to="/questions/2"
      >
        Another discussion
      </Link>
    </h3>
    <p
      className="user_name"
    >
      <span>
        <Link
          replace={false}
          to="/user/2"
        >
          Bodeep Deboop
        </Link>
      </span>
        
      <span>
        b.deboop
      </span>
    </p>
    <div
      className="discussion_info"
    >
      <div
        className="Tags"
      >
        <span
          className="tag"
          key="javascript"
        >
          javascript
           
        </span>
        <span
          className="tag"
          key="react"
        >
          react
           
        </span>
      </div>
      <div>
        <span>
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            fromNow={true}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-03-27T20:27:19.481Z
          </t>
        </span>
           
        <span>
          1
           respond(s)
        </span>
      </div>
    </div>
  </div>
</Fragment>
`;
